Plugin crashes on unknown fields in the API responses

FATAL: .Problem with reading the response message, class : class com.qubell.services.ws.LaunchInstanceResponse, ContentType : application/json;charset=utf-8.
javax.ws.rs.client.ClientException: .Problem with reading the response message, class : class com.qubell.services.ws.LaunchInstanceResponse, ContentType : application/json;charset=utf-8.
	at org.apache.cxf.jaxrs.client.AbstractClient.reportMessageHandlerProblem(AbstractClient.java:702)
	at org.apache.cxf.jaxrs.client.AbstractClient.readBody(AbstractClient.java:475)
	at org.apache.cxf.jaxrs.client.WebClient.handleResponse(WebClient.java:971)
	at org.apache.cxf.jaxrs.client.WebClient.doResponse(WebClient.java:955)
	at org.apache.cxf.jaxrs.client.WebClient.doChainedInvocation(WebClient.java:905)
	at org.apache.cxf.jaxrs.client.WebClient.doInvoke(WebClient.java:772)
	at org.apache.cxf.jaxrs.client.WebClient.doInvoke(WebClient.java:759)
	at org.apache.cxf.jaxrs.client.WebClient.invoke(WebClient.java:355)
	at org.apache.cxf.jaxrs.client.WebClient.post(WebClient.java:381)
	at com.qubell.services.ws.WebServiceBase.invoke(WebServiceBase.java:159)
	at com.qubell.services.ws.ApplicationServiceWsImpl.launch(ApplicationServiceWsImpl.java:61)
	at com.qubell.services.QubellFacadeImpl.launchInstance(QubellFacadeImpl.java:74)
	at com.qubell.jenkinsci.plugins.qubell.builders.StartInstanceBuilder.perform(StartInstanceBuilder.java:218)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:756)
	at hudson.model.Build$BuildExecution.build(Build.java:198)
	at hudson.model.Build$BuildExecution.doRun(Build.java:159)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:529)
	at hudson.model.Run.execute(Run.java:1706)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:232)
Caused by: org.codehaus.jackson.map.exc.UnrecognizedPropertyException: Unrecognized field "warnings" (Class com.qubell.services.ws.LaunchInstanceResponse), not marked as ignorable
 at [Source: sun.net.www.protocol.http.HttpURLConnection$HttpInputStream@72957363; line: 1, column: 46] (through reference chain: com.qubell.services.ws.LaunchInstanceResponse["warnings"])
	at org.codehaus.jackson.map.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:53)
	at org.codehaus.jackson.map.deser.StdDeserializationContext.unknownFieldException(StdDeserializationContext.java:267)
	at org.codehaus.jackson.map.deser.std.StdDeserializer.reportUnknownProperty(StdDeserializer.java:673)
	at org.codehaus.jackson.map.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:659)
	at org.codehaus.jackson.map.deser.BeanDeserializer.handleUnknownProperty(BeanDeserializer.java:1365)
	at org.codehaus.jackson.map.deser.BeanDeserializer._handleUnknown(BeanDeserializer.java:725)
	at org.codehaus.jackson.map.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:703)
	at org.codehaus.jackson.map.deser.BeanDeserializer.deserialize(BeanDeserializer.java:580)
	at org.codehaus.jackson.map.ObjectMapper._readValue(ObjectMapper.java:2704)
	at org.codehaus.jackson.map.ObjectMapper.readValue(ObjectMapper.java:1315)
	at org.codehaus.jackson.jaxrs.JacksonJsonProvider.readFrom(JacksonJsonProvider.java:419)
	at org.apache.cxf.jaxrs.utils.JAXRSUtils.readFromMessageBodyReader(JAXRSUtils.java:1184)
	at org.apache.cxf.jaxrs.client.AbstractClient.readBody(AbstractClient.java:471)
	... 20 more

--- a/pom.xml
+++ b/pom.xml
@@ -26,7 +26,7 @@
     <groupId>com.qubell.jenkins-ci.plugins</groupId>
     <artifactId>qubell</artifactId>
     <name>Qubell Plugin</name>
-    <version>2.5</version>
+    <version>2.6</version>
     <packaging>hpi</packaging>
 
     <developers>

--- a/src/main/java/com/qubell/services/InstanceStatus.java
+++ b/src/main/java/com/qubell/services/InstanceStatus.java
@@ -163,22 +163,23 @@ public class InstanceStatus {
 
         InstanceStatus that = (InstanceStatus) o;
 
-        return application.equals(that.application)
-                && currentWorkflow == null ? that.currentWorkflow == null : currentWorkflow.equals(that.currentWorkflow)
-                && instance.equals(that.instance)
-                && returnValues == null ? that.returnValues == null : returnValues.equals(that.returnValues)
-                && status == that.status
-                && version.equals(that.version);
+        return (application == null ? that.application == null : application.equals(that.application))
+               && (currentWorkflow == null ? that.currentWorkflow == null : currentWorkflow.equals(that.currentWorkflow))
+               && (instance == null ? that.instance == null : instance.equals(that.instance))
+               && (returnValues == null ? that.returnValues == null : returnValues.equals(that.returnValues))
+               && status == that.status
+               && (version == null ? that.version == null : version.equals(that.version));
+
     }
 
     @Override
     public int hashCode() {
-        int result = instance.hashCode();
-        result = 31 * result + version.hashCode();
-        result = 31 * result + status.hashCode();
+        int result = instance != null ? instance.hashCode() : 0;
+        result = 31 * result + (version != null ? version.hashCode() : 0);
+        result = 31 * result + (status != null ? status.hashCode() : 0);
         result = 31 * result + (currentWorkflow != null ? currentWorkflow.hashCode() : 0);
         result = 31 * result + (returnValues != null ? returnValues.hashCode() : 0);
-        result = 31 * result + application.hashCode();
+        result = 31 * result + (application != null ? application.hashCode() : 0);
         return result;
     }
 }

--- a/src/main/java/com/qubell/services/ws/LaunchInstanceResponse.java
+++ b/src/main/java/com/qubell/services/ws/LaunchInstanceResponse.java
@@ -18,6 +18,7 @@ package com.qubell.services.ws;
 
 import javax.xml.bind.annotation.XmlRootElement;
 import javax.xml.bind.annotation.XmlType;
+import java.util.List;
 
 /**
  * Response of instance launch WS call
@@ -28,6 +29,10 @@ import javax.xml.bind.annotation.XmlType;
 public class LaunchInstanceResponse {
 
     private String id;
+
+    private List<String> warnings;
+
+    private List<String> errors;
 
     /**
      * Gets instance id
@@ -44,4 +49,21 @@ public class LaunchInstanceResponse {
     public void setId(String id) {
         this.id = id;
     }
+
+    public List<String> getWarnings() {
+        return warnings;
+    }
+
+    public void setWarnings(List<String> warnings) {
+        this.warnings = warnings;
+    }
+
+    public List<String> getErrors() {
+        return errors;
+    }
+
+    public void setErrors(List<String> errors) {
+        this.errors = errors;
+    }
+
 }

